bigquery/snippets/managedwriter/bigquerystorage_append_rows_default.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package managedwriter
+
+// [START bigquerystorage_write_default_complexschema]
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/rand"
+	"time"
+
+	"cloud.google.com/go/bigquery/storage/managedwriter"
+	"cloud.google.com/go/bigquery/storage/managedwriter/adapt"
+	"github.com/GoogleCloudPlatform/golang-samples/bigquery/snippets/managedwriter/exampleproto"
+	"google.golang.org/protobuf/proto"
+)
+
+// generateExampleMessages generates a slice of serialized protobuf messages using a statically defined
+// and compiled protocol buffer file, and returns the binary serialized representation.
+func generateExampleDefaultMessages(numMessages int) ([][]byte, error) {
+	msgs := make([][]byte, numMessages)
```

Comment:
nit: I don't think you need to create an array here. You can create a variable-length slice and append to it later on. You don't even need to use `make()`!

```
var msgs [][]byte

// ... stuff happens
msgs.append(msgs, b)
```

Totally up to you.

Reply:
its a minor difference, but in this case we know exactly the size we need so we avoid unneeded allocations by specifying explicitly.